app/controllers/page_mount_controller.rb

```diff
@@ -50,13 +52,19 @@ def domains_layout
 
   private
 
-  def store_client_type_id_in_cms_session(proxy_response)
+  def store_client_type_id_in_cms_session(response, request)
     # Special Hack so CC can show exclusive products for clients
-    session[:client_type_id] = JSON.parse(proxy_response.body)['client_type_id']
-  end
-
-  def login_details_request?(request)
-    request.path == '/parties/login_header_details'
+    case request.path
+    when '/parties/login.json'
+      json = JSON.parse(response.body)
+      session[:client_id] = json['client_id']
```

Comment:
The storage is up to the CMS including this extension, cc and kis use redis for sessions so we're safe.

Also of note is we're already doing this with 1 variable, now it's 3 so if we had an issue it should have already surfaced.